### DISCUSSION TITLE
Copilot not working on code OSS anymore

### DISCUSSION THREAD

**Discussion by jon4hz** [community/community#11509] [Copilot Conversations]
Recently github copilot stopped working on code oss. 
Error: `GitGub Copilot requires VS Code 1.57+` even though I'm on version 1.64.0.

Do we really have to switch proprietary software in order to use the copilot?

**Comment by kali-37** [comment]
### Add following lines on 

 - .vscode-oss/argv.json
			
		// Allows to disable crash reporting.
		// Should restart the app if the value is changed.
		"enable-crash-reporter": true,
		"password-store": "gnome",	
		// Unique id used for correlating crash reports sent from this instance.
		// Do not edit this value.
		"crash-reporter-id": "4805a1ab-42bd-4ad7-befc-3c4c84ac6a81",
		"enable-proposed-api": ["github.copilot"]

Added
 1. ["enable-proposed-api": ["github.copilot"] ] 
 2. [ "password-store": "gnome"] 



**$ cat /etc/pam.d/login**

			#%PAM-1.0
			auth       required     pam_securetty.so
			auth       requisite    pam_nologin.so
			auth       include      system-local-login
	[+]		auth       optional     pam_gnome_keyring.so
			account    include      system-local-login
			session    include      system-local-login
			password   include      system-local-login
	[+]		session    optional     pam_gnome_keyring.so auto_start


**Added** [+] above . auto start gnome_keyring.so on login 


**Make sure :** 

 1. You have installed gnome-keyring
 2. You have VSIX MANAGER extension on code-oss 
	
	After installing VSIX Manager , 
		
		Download .vsix file of githubcopilot extension 
		
	[extension link ..](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot)
		
		Now install .vsix file navigating it using Ctrl+shift+P ,
		command :  **Install From VSIX**   then select the .vsix file.

Source Added : [Find more ](https://github.com/microsoft/vscode/issues/187338)

**Comment by trinitronx** [comment]
> Recently github copilot stopped working on code oss. Error: `GitGub Copilot requires VS Code 1.57+` even though I'm on version 1.64.0.
> 
> Do we really have to switch proprietary software in order to use the copilot?

Unfortunately, this has been an issue for many *nix distributions other than Ubuntu + compatible Debian-based ones, and RedHat + compatible RHEL-based ones for quite some time.

## TLDR;

There is a split between VSCode + Microsoft Extension Marketplace (official VSCode), and the Open Source VSCodium (a.k.a. `Code OSS`) + OpenVSX Extension Marketplace.  It's unclear whether this split is truly necessary, given the unclear legal stance from Microsoft, and the community's general abundance of caution (possibly with a lot of fear, uncertainty and doubt mixed in).  In any case, it is now difficult for many *nix redistributed VSCodium versions to work out of the box with the `.vsix` Copilot extensions.  This is due to heavy development and API Proposals changing rapidly, so compatibility between versions of GitHub Copilot extensions and VSCode itself (including enabled API proposals in `product.json`) seems to have been difficult for the community to keep up with.

Some distributions such as Arch Linux and other similar `pacman`-based distros can install the [`code-features`][2] package from the AUR.  However, due to the API Proposals changing, it may be necessary to [manually patch the `product.json`][4] to enable the new API Proposals that GitHub Copilot extensions need to work.

Finally, recent versions of both GitHub Copilot extensions have started using a more constrictive version constraint on the version of VSCode or VSCodium that they will install on.  In practice, it seems that this version constraint is overly cautious and the extensions generally work fine on versions of VSCodium of at least `1.90.x` in most users experience.  So, you may also need to [unpack the `.vsix` zip files for Copilot extensions, and manually edit the `package.json` inside to relax the version constraint][5].

## The Detailed Version

Microsoft currently only distributes officially the `.deb`, `.rpm`, and `.tar.gz` VSCode packages and tarball, leaving many other *nix distributions and packaging systems in the dust.  As such, many distributions have had to either repackage the official versions such that the native package manager can be used, or simply distribute a version of VSCodium (a.k.a. `Code OSS`).

For those *nix distributions that chose to package VSCodium, there are other concerns regarding the Extensions Marketplace.  There has been a bit of confusion given the quasi-OSS status of official VSCode versus the fully OSS VSCodium, and some rather vague and unclear legal stances and interpretations regarding the use of Microsoft's official Extension Marketplace by the OSS VSCodium and possibly other ports to unsupported operating systems.  This has caused a split between the official Extensions Marketplace, and the [OpenVSX Marketplace][1] which is maintained by the Eclipse Foundation.  Additionally, because `GitHub.copilot` and `GitHub.copilot-chat` extensions are not fully Open Source, these particular extensions have only been available from Microsoft's Extension Marketplace.

Users of VSCodium have had to install the `.vsix` extension, as you're probably well aware.  Recently, it has become tricky to install in VSCodium due to the tight coordination required between distributing VSCode itself and its enabled API Proposals (found in the `product.json`), alongside the GitHub Copilot extensions and their corresponding APIs, which have both been in flux due to being under heavy development.  This has resulted in users of VSCodium being left in the situation of the original poster here, @jon4hz, where the extension seems to break after any upgrade of VSCode and/or the `.vsix` Copilot extensions.  This is mostly due to an oversight (or perhaps intentional omission) when it comes to packaging and distributing VSCodium with a custom `product.json`, which can make it incompatible with the GitHub Copilot extension's API proposals if it does not include these.

As such, users of VSCodium have had to patch the `product.json` and keep a close eye on any new or deprecated API proposals for both `GitHub.copilot` and `GitHub.copilot-chat` extensions each and every time that a version upgrade happens for all three of these pieces of software.

It's really no different from the state of VSCode, as evidenced by microsoft/vscode#214294 regarding the discussion of packaging and distributing VSCode in lockstep with GitHub Copilot extensions.  In the case of the official VSCode, the users of the official package benefit from the work done by the official packagers.  Meanwhile, in the case of VSCodium, `Code OSS` and other repackaged OSS versions of VSCode, users are left either in the dark unable to use the Copilot extensions, or with a `product.json` patching process that must be maintained manually.

Luckily, some of this work has been done by Arch Linux community members in the [`code-features` package][2].  If you are on an Arch-based distro, you might try installing that package alongside the main `code` (a.k.a. `Code OSS`) package.  It handles patching the `product.json` in a few useful ways, but sometimes does not always keep in lockstep with the latest API proposals from GitHub Copilot.  So, you still may need to continue [patching it or manually editing it yourself][4] anytime an API change happens.

Finally, some recent versions of the two Copilot extensions have had version constraints specified for which VSCode version they officially support.  If you're familiar with Node.js `package.json` this will be no suprise, as it's common to lock dependency versions for `npm` packages for compatibility reasons.  However, unfortunately for VSCodium users, the version constraints in `package.json` can be a bit too constrictive and prevent the extension from installing and working with the version of VSCodium that is generally available in their OS package manager.  As such, it may be necessary to also unpack the `.vsix` zip file, and [manually edit the `package.json` for GitHub Copilot Extensions][5] in addition to the VSCode `product.json` mentioned above.  Do note that this may result in some buggy behavior cropping up, but at least it will get the extensions working with the version of VSCodium available to you.  In my experience so far, there hasn't been any noticeable issues that I've encountered with any kind of obvious incompatibility.

Hopefully, given enough time, the GitHub Copilot API will stabilize (it seems they [announced some level of "stable" availability recently][3]), and VSCodium may be able to work out of the box without patching `product.json` all the time.

I hope this gives you some much needed context about the current situation.  It's not ideal, and there are some grey areas and need for communication and clarity when it comes to the VSCode vs. OSS / VSCodium usage of Microsoft's Extension Marketplace, the GitHub Copilot extensions, and Microsoft's general intent towards unsupported *nix distributions.

I'm hanging onto hope that one day, these problems will all clear up, and we can all happily coexist with whatever version of VSCode and/or Extension Marketplace works on our operating system of choice.

[1]: https://open-vsx.org/
[2]: https://aur.archlinux.org/packages/code-features
[3]: https://code.visualstudio.com/blogs/2024/06/24/extensions-are-all-you-need#_available-today-to-everyone
[4]: https://github.com/VSCodium/vscodium/issues/1546#issuecomment-1773179869
[5]: https://github.com/VSCodium/vscodium/issues/1546#issuecomment-2211798305

**Comment by kay-is** [comment]
I'm using Manjaro/Arch and installed `code-features` and `code-marketplace` via AUR.

Then I installed the GitHub Copilot extensions and logged into GitHub. I didn't get any notification to do so, I had to click the user icon on the left and the menu would show that Copilot wasn't logged in.

I wouldn't see any additional extension icons on the left, but the `ctrl+p` commands to open the Copilot chat window worked and auto-complete worked too.
